txstratum/manager.py

```diff
@@ -40,6 +40,9 @@ class TxMiningManager:
     DEFAULT_BLOCK_TEMPLATE_UPDATE_INTERVAL = 3.0  # seconds
     TX_CLEAN_UP_INTERVAL = 300.0  # seconds
 
+    MAX_RETRY_FOR_FAILED_PUSH_TX = 3
+    RETRY_WAIT_FOR_FAILED_PUSH_TX = 3  # seconds
```

Comment:
Maybe this is too slow? This will affect the time a job takes to be processed right?

Reply:
Yeah, if we consider that "processing" the job includes pushing the transaction (if they have propagate=true), then yes, we would increase the time.

For wallets that use the tx-mining-service, I think they wouldn't be affected by this, because they propagate the transactions themselves, not on tx-mining-service (maybe I should confirm this).

But the reason why I didn't set a lower time is to give the fullnode some time to recover before trying again, if the error was just due to temporary load.

What do you think?

Reply:
If the default is for wallets to push the transactions themselves, I think a retry mechanism is really not that relevant, if it fails it fails and we move on, blocks found by the tx-mining-service aren't urgently important, and in the few cases they are actually needed (like the testnet), they will be mined and found again, just at a later timestamp.

But for the cases that opt to let the tx-mining-service push the transactions, I'd say it is unexpected for a job to succeed and the tx not be pushed, so maybe a better behavior would be to wait for the push of the transaction to mark the job as complete.